lib/internal/url.js

```diff
@@ -196,16 +188,12 @@ function onParsePathComplete(flags, protocol, username, password,
 
 function onParseSearchComplete(flags, protocol, username, password,
                                host, port, path, query, fragment) {
-  if (flags & binding.URL_FLAGS_FAILED)
-    return;
```

Comment:
I'm also curious about it. Is that why the search setter doesn't need to have `onParseError` in this case?